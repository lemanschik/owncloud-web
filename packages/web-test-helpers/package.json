{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
