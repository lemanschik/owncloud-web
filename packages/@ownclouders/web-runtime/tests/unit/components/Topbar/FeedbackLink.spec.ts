import FeedbackLink from '../../../../src/components/Topbar/FeedbackLink.vue'
import { axe, toHaveNoViolations } from 'jest-axe'
import { defaultPlugins, mount } from 'web-test-helpers'

expect.extend(toHaveNoViolations)

describe('FeedbackLink component', () => {
  // FIXME: skip flaky test, also see: https://github.com/owncloud/web/issues/8244
  it.skip('has no accessibility violations', async () => {
    const { wrapper } = getWrapper()
    expect(
      await axe(wrapper.html(), {
        rules: {
          region: { enabled: false }
        }
      })
    ).toHaveNoViolations()
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('properties', () => {
    it('allows to overwrite the link href', async () => {
      const { wrapper } = getWrapper()
      const url = 'https://some-link.tld/'
      await wrapper.setProps({ href: url })
      expect(wrapper.find('a').attributes().href).toEqual(url)
    })
    it('allows to overwrite the link ariaLabel', async () => {
      const { wrapper } = getWrapper()
      const ariaLabel = 'some aria label'
      await wrapper.setProps({ ariaLabel })
      expect(wrapper.find('a').attributes()['aria-label']).toEqual(ariaLabel)
    })
    it('allows to overwrite the link description', async () => {
      const { wrapper } = getWrapper()
      const description = 'some lengthy description'
      await wrapper.setProps({ description })
      expect(wrapper.find('#oc-feedback-link-description').text()).toEqual(description)
    })
  })
})

const getWrapper = () => {
  return {
    wrapper: mount(FeedbackLink, {
      global: {
        plugins: [...defaultPlugins()]
      }
    })
  }
}
