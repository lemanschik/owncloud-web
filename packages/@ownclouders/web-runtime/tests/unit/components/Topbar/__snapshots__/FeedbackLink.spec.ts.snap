// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackLink component has no accessibility violations 1`] = `
<div class="oc-flex">
  <a aria-describedby="oc-feedback-link-description" aria-label="ownCloud feedback survey" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-inverse oc-button-inverse-raw" href="https://owncloud.com/web-design-feedback" target="_blank">
    <!-- @slot Content of the button -->
    <span class="oc-icon oc-icon-m oc-icon-passive">
      <!---->
    </span>
  </a>
  <p class="oc-invisible-sr" id="oc-feedback-link-description">Provide your feedback: We'd like to improve the web design and would be happy to hear your feedback. Thank you! Your ownCloud team.</p>
</div>
`;
